otp4gb/routing.py

```diff
@@ -235,39 +249,40 @@ def add_error(message: str) -> None:
     retries = 0
     error_message = []
     end = False
-    while True:
+
+    with requests.Session() as session:
         req = requests.Request("GET", url, params=parameters.params())
-        prepared = req.prepare()
-
-        try:
-            session = requests.Session()
-            response = session.send(prepared, timeout=REQUEST_TIMEOUT)
-        except requests.exceptions.RequestException as error:
-            msg = f"{error.__class__.__name__}: {error}"
-            add_error(msg)
-            response = _FakeResponse(url=prepared.url, status_code=-10, reason=msg)
-
-        if response.status_code == requests.codes.OK:
-            result = RoutePlanResults.parse_raw(response.text)
-            if result.error is None:
+        prepared = session.prepare_request(req)
+
+        while True:
+            try:
+                response = session.send(prepared, timeout=REQUEST_TIMEOUT)
+            except requests.exceptions.RequestException as error:
+                msg = f"{error.__class__.__name__}: {error}"
+                add_error(msg)
+                response = _FakeResponse(url=prepared.url, status_code=-10, reason=msg)
+
+            if response.status_code == requests.codes.OK:
+                result = RoutePlanResults.model_validate_json(response.text)
+                if result.error is None:
+                    return response.url, result
+                if result.error.id in OTP_ERRORS.values():
+                    return response.url, result
+
+                add_error(
+                    f"OTP Error {result.error.id}: {result.error.msg} {result.error.message}"
+                )
+
+            if end or retries > REQUEST_RETRIES:
```

Comment:
is end not alawys False here, 

Is there something to be added later that may use `end` ?

Reply:
Yes, you're right I'm not sure exactly if end was previously used for something that was removed or not but I don't think it's needed anymore

Reply:
You're correct, I can't remember if `end` was previously used for something that has since been removed or not but its definitely not needed now. I'll remove it